File: navigation/src/main/java/com/example/navigation/stepsEngine/field/PaymentMethodIdField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class PaymentMethodIdField extends Field {

    @Override
    public String get(FlowState flowState) {
        return flowState.getPaymentMethodWrapper().getPaymentMethodId().name();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/PaymentMethodIdField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState

class PaymentMethodIdField : Field() {
    override operator fun get(flowState: FlowState): String? {
        return flowState.paymentMethodWrapper.paymentMethodId?.name
    }

    override fun getId(): String {
        return FieldName.PAYMENT_METHOD_ID
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}